src/luautils.h

```diff
@@ -0,0 +1,73 @@
+
+#pragma once
+
+#include <string>
+#include "lua/src/lua.hpp"
+
+using namespace std;
+
+// utilities for lua stack manipulation
+struct LuaStackPtr{
+	LuaStackPtr(const int idx_):idx(idx_){}
+	const int idx;
+};
+
+struct StringPtr{
+	StringPtr(const char* c_, const size_t len_):c(c_),len(len_){}
```

Comment:
I think a constructor taking a `std::string` would be beneficial for this class:
```c++
explicit StringPtr(const std::string& s)
: c(s.data())
, len(s.size())
{}
```

Reply:
Main purpose of this class is to hand c-strings to the overloaded lua_push function as single arguments. As there is also a lua_push that takes std::strings, the StringPtr would not be needed there. Do y... Would there still be a use for the suggested constructor? To make StringPtr more general maybe?

Reply:
It was intended for the call `lua_pushfield(L, rule)` in `main.cpp` line 92, but by the time I got there I realised it could just pass the `std::string`. So *you're* right, it would actually be confusing to have a `std::string` c'tor and this `struct` should stay as it is ;-)